backend/FwLite/MiniLcm/SyncHelpers/EntrySync.cs

```diff
@@ -162,27 +171,88 @@ public override Task<int> Replace(ComplexFormComponent beforeComponent, ComplexF
         }
     }
 
-    private class SensesDiffApi(IMiniLcmApi api, Guid entryId) : IOrderableCollectionDiffApi<Sense>
+    private class ComplexFormComponentsDiffApi(Entry afterEntry, IMiniLcmApi api) : IOrderableCollectionDiffApi<ComplexFormComponent>
+    {
+        private readonly bool supportsEntityIds = api.GetDataFormat() == ProjectDataFormat.Harmony;
+
+        public Guid GetId(ComplexFormComponent component)
+        {
+            // we can't use the ID as there's none defined by Fw so it won't work as a sync key
+            return component.ComponentSenseId ?? component.ComponentEntryId;
+        }
+
+        private BetweenPosition MapBackToEntityIds(BetweenPosition between)
+        {
+            var previous = between!.Previous is null ? null : afterEntry.Components.Find(c => GetId(c) == between.Previous);
+            var next = between!.Next is null ? null : afterEntry.Components.Find(c => GetId(c) == between.Next);
+            return new BetweenPosition(previous?.Id, next?.Id);
+        }
+
+        public async Task<int> Add(ComplexFormComponent after, BetweenPosition between)
+        {
+            if (supportsEntityIds) between = MapBackToEntityIds(between);
+
+            //change id, since we're not using the id as the key for this collection
+            //the id may be the same, which is not what we want here
+            after.Id = Guid.NewGuid();
+            try
+            {
+                await api.CreateComplexFormComponent(after, between);
+            }
+            catch (NotFoundException)
+            {
+                //this can happen if the entry was deleted, so we can just ignore it
+            }
+            return 1;
+        }
+
+        public async Task<int> Move(ComplexFormComponent component, BetweenPosition between)
+        {
+            if (supportsEntityIds) between = MapBackToEntityIds(between);
+            var componentId = supportsEntityIds ? component.Id : component.ComponentSenseId ?? component.ComponentEntryId;
+            await api.MoveComplexFormComponent(afterEntry.Id, componentId, between);
```

Comment:
yeah this won't work. Imagine a client (fw lite, or FLExTools) calling MoveComplexFormComponent directly. What does it pass in to the 2nd arg? component.Id? or component.ComponentEntryId? It depends on GetDataFormat, that really sucks. See my comment above for how we might fix this